lore/estimators/sklearn.py

```diff
@@ -63,6 +65,23 @@ class Regression(Base):
 
 
 class BinaryClassifier(Base):
+    def __init__(self, estimator):
+        super(BinaryClassifier, self).__init__(estimator, eval_metric='logloss', scoring_metric='auc')
+
+    @before_after_callbacks
+    @timed(logging.INFO)
+    def evaluate(self, x, y):
+        import sklearn
+        y_pred = self.predict_proba(x)
+        return sklearn.metrics.log_loss(y, y_pred)
+
+    @before_after_callbacks
+    @timed(logging.INFO)
+    def score(self, x, y):
+        import sklearn
+        y_pred = self.predict_proba(x)[:, 1]
```

Comment:
Can we grab some time to rationalize predict_proba vs predict. The reasoning has gotten fuzzy for me.

Reply:
`predict_proba` should only be available for some classifiers that also output a score. A lot of `sklearn` classifiers have a `predict` method and a `predict_proba` method if applicable. Same with `xgboost`.